examples/gallery/plot/scatter3d.py

```diff
@@ -0,0 +1,52 @@
+"""
+3D Scatter plots
+----------------
+
+The :meth:`pygmt.Figure.plot3d` method can be used to plot symbols in 3D.
+In the example below, we show how the Iris flower dataset can be visualized
+using a perspective 3-dimensional plot. The ``region`` argument has to be set
+to include the :math:`x`, :math:`y`, :math:`z` axis limits in the form of
+(xmin, xmax, ymin, ymax, zmin, zmax), and this can be done automatically using
+:meth:`pygmt.info`. To include the z-axis stick, ``frame`` needs to be set as a
+minimum to something like ``frame=["WsNeZ", "zaf"]``. Use ``perspective`` to
+control the azimuth and elevation angle of the view, and ``zscale`` to adjust
+the vertical exaggeration factor.
+"""
+
+import pandas as pd
+import pygmt
+
+# Load sample iris data, and convert 'species' column to categorical dtype
+df = pd.read_csv("https://github.com/mwaskom/seaborn-data/raw/master/iris.csv")
+df["species"] = df.species.astype(dtype="category")
+
+# Use pygmt.info to get region bounds (xmin, xmax, ymin, ymax, zmin, zmax)
+# The below example will return a numpy array like [0., 3., 4., 8., 1., 7.]
+region = pygmt.info(
+    table=df[["petal_width", "sepal_length", "petal_length"]],  # x, y, z columns
+    per_column=True,  # report output as a numpy array
+    spacing="1/2/0.5",  # rounds x, y and z intervals by 1, 2 and 0.5 respectively
+)
+
+# Make our 3D scatter plot, coloring each of the 3 species differently
+fig = pygmt.Figure()
+pygmt.makecpt(cmap="cubhelix", series=(0, 3, 1))
```

Comment:
This `makecpt` command produces a discrete CPT (see the figure below). It doesn't make sense for the categorical data here. 
![image](https://user-images.githubusercontent.com/3974108/97262849-17968200-17f8-11eb-9f4b-0c22ed1944a8.png). 

Instead, we should use `makecpt -F+c`, which produces categorical CPT.

![image](https://user-images.githubusercontent.com/3974108/97263175-c2a73b80-17f8-11eb-9d04-f019721baddd.png)

But the colorbar annotations don't make sense to me. Perhaps another GMT bug.

Note: the gridlines are not shown in GMT 6.1.1. It's a new future in GMT master (see https://github.com/GenericMappingTools/gmt/pull/3993), but the gridlines in XZ and YZ planes may also have bugs. Will report to GMT later.

Reply:
Already reported to upstream https://github.com/GenericMappingTools/gmt/issues/4387 and https://github.com/GenericMappingTools/gmt/issues/4388.

Reply:
> Instead, we should use makecpt -F+c, which produces categorical CPT.
> But the colorbar annotations don't make sense to me. Perhaps another GMT bug.

Ah yes, forgot about [`-F+c`](https://docs.generic-mapping-tools.org/latest/makecpt.html#f)! Actually for the Iris species categories are nominal (discrete) rather than ordinal (i.e. ranked 1st, 2nd, 3rd), so a legend would make more sense here than a colorbar. Another thought I had was to label each species using `text` plotted in 3D, but we haven't implemented that 'feature' yet.

So for now, I think we can just use `-F+c` here and ignore the colorbar, though we should add another categorical gallery example to close #244? I'd like to avoid adding a legend since it requires a for-loop to make the labels (and this complicates the example).

The wall frames look good though, too bad we need to wait for GMT 6.2.0! But good that we found two bugs to report to upstream GMT :laughing:

Reply:
Actually, `makecpt -F` is not aliased yet. Do we want to do that here or in a separate PR?

Reply:
Alright, with #676 merged, I've updated the code to use `pygmt.makecpt(..., color_model="+c", ...)`

https://github.com/GenericMappingTools/pygmt/blob/40f7120cc20dc96aac8ab8d431dfd4b124274b6e/examples/gallery/plot/scatter3d.py#L34

In the future when we bump the minimum GMT version to 6.2.0 which would include the new feature at https://github.com/GenericMappingTools/gmt/pull/4390, we can do the following:

```pygmt
pygmt.makecpt(cmap="cubhelix", color_model="+cSetosa,Versicolor,Virginica", series=(0, 3, 1))
fig.plot3d(...)
fig.colorbar()
fig.show()
```

which would produce a nice colorbar ;)